fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/AggregateIndexExpansionVisitor.java

```diff
@@ -191,8 +192,21 @@ private Pair<Quantifier, List<Placeholder>> constructSelectWhereAndPlaceholders(
         }
 
         // add an RCV column representing the grouping columns as the first result set column
+        // also, make sure to set the field type names correctly for each field value in the grouping keys RCV.
         final var groupingValue = RecordConstructorValue.ofColumns(
-                baseExpansion.getResultColumns().subList(0, groupingKeyExpression.getGroupingCount()));
+                baseExpansion.getResultColumns().subList(0, groupingKeyExpression.getGroupingCount())
+                        .stream()
+                        .map(column -> {
+                            if (column.getValue() instanceof FieldValue) {
```

Comment:
I thought there was some helper that would allow you to inherit the last field name in the access path of a field value to become the name of the column if that's the input to an RCV(). There is definitely a method to inquire of a field value's last step; maybe that will help find the method I am thinking of.

Reply:
If that logic does not already exist it would be useful to add it in RCV if possible so other code can benefit from that.

Reply:
I wasn't aware of that, checking.

Reply:
I did some investigation and tried to move this field name amendment logic to `RecordConstructorValue#ofColumns` to become the following:

```java
    @Nonnull
    private static Type.Record computeResultType(@Nonnull final Collection<Column<? extends Value>> columns) {
        final var fields = columns.stream()
                .map(column -> {
                    if (column.getValue() instanceof FieldValue) {
                        final var fieldValueNameMaybe = ((FieldValue)column.getValue()).getLastFieldName();
                        return fieldValueNameMaybe.map(fieldValueName -> column.getField().withName(fieldValueName)).orElse(column.getField());
                    }
                    return column.getField();
                })
                .collect(ImmutableList.toImmutableList());
        return Type.Record.fromFields(false, fields);
    }
```

This looks innocent, however doing so broke too many tests that effectively try to do something like:

```sql
select a, a, a, a, a from t;
```

Because the now-duplicate field names are rejected by `Type#normalizeFields` which [guards](https://github.com/FoundationDB/fdb-record-layer/blob/f2ac2b9b35f5f679307cfe2389f75552f294878e/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java#L1975) against this.

It is interesting to see _why_ this works, it is because the `ExpansionVisitor` takes care of [deduping](https://github.com/FoundationDB/fdb-record-layer/blob/f2ac2b9b35f5f679307cfe2389f75552f294878e/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/GraphExpansion.java#L177) field names, before they are handed in to `Type#normalize` later on during plan generation.

Reply:
So, I think it is ok to leave it localised here?

Reply:
yup